spaceship/lib/spaceship/tunes/iap_list.rb

```diff
@@ -58,6 +58,12 @@ def status
       def edit
         attrs = client.load_iap(app_id: application.apple_id, purchase_id: self.purchase_id)
         attrs[:application] = application
+
+        if attrs["addOnType"] == Spaceship::Tunes::IAPType::RECURRING
+          attrs["pricingIntervals"] = client.load_recurring_iap_pricing(app_id: application.apple_id,
```

Comment:
* We could do this lazily inside of `IAPDetail#pricing_intervals`
* could we avoid throwing away the rest of the response and store it in an attribute called like `raw_pricing_data`? (I think this is reasonably within scope of this PR as opposed to fully dealing with introductory pricing)

Reply:
1. I suggest against it, `IAPDetail#pricing_intervals` is a simple getter, and if I were to fetch it lazily it would have to make an HTTP request which will take longer. `product.edit` makes a request anyway so from that perspective the request should take only a while longer.
2. It's a good idea, but it makes things a bit more difficult:

- I have to maintain both this and `raw_data["pricingIntervals"]` when the user changes the pricing intervals, and whenever `raw_data["pricingIntervals"]` is accessed, I might have to use `raw_pricing_data["subscriptions"]` make it so that `raw_pricing_data["subscriptions"]` is used instead.
- `raw_pricing_data` will have to go into `IAPDetail`, and it has to be public even thought its kinda implementation detail. I would have added it as a private field and initialize it once in `initialize`, but I can't because the `Base` class takes over the initializer.

Then I thought that `raw_pricing_data` can be put inside `raw_data` itself. But since `update_iap!` uses it as a whole, it will send `raw_pricing_data` which will cause the request to fail. So I have to remove it from `raw_data` before `update_iap!` is called. What do you think of this solution?

Reply:
That's why I suggested moving the entire logic into IAPDetails, such that it would fetch pricing intervals and introductory offers on its own, and store *both* only in `raw_pricing_data` (not adding keys to `raw_data` at all)

I can make a separate PR for this if you want

>it has to be public even thought its kinda implementation detail

this is crucial, "we" (our company) need access to the raw data as long as introductory offers are not fully implemented.

Reply:
After some thought I agree with you, we should not add keys to `raw_data` because it is used in many places and sending a product update request with the new keys will fail.
So I added a `raw_pricing_data` field to `IAPDetail`. I initialize it from the pricing data in `raw_data` that I add beforehand, and then remove the superfluous data from `raw_data`.